nix/searxng.nix

```diff
@@ -0,0 +1,94 @@
+{ pkgs
+, lib
+, name
+, config
+, ...
+}:
+let
+  inherit (lib) types;
+
+  settingType =
+    with types;
+    (oneOf [
+      bool
+      int
+      float
+      str
+      (listOf settingType)
+      (attrsOf settingType)
+    ])
+    // {
+      description = "JSON value";
+    };
+in
+{
+  options = {
+    enable = lib.mkEnableOption name;
+    package = lib.mkPackageOption pkgs "searxng" { };
+
+    host = lib.mkOption {
+      description = "Searxng bind address";
+      default = "127.0.0.1";
+      type = types.str;
+    };
+
+    port = lib.mkOption {
+      description = "Searxng port to listen on";
+      default = 1234;
+      type = types.port;
+    };
+
+    settings = lib.mkOption {
+      type = types.attrsOf settingType;
+      default = {
+        use_default_settings = true;
+        server.secret_key = "secret";
+        server.limiter = false;
+      };
```

Comment:
I tried out the defaults, seems like `use_default_settings` and `server.secret_key` are essential defaults to start the server. How about we move them to where `server.{bind_address, port}` are being set? 

This way we could keep the default as `{ }`, allowing the user to override the settings without copying over the defaults. 

Also, about `server.limiter = false;`, I didn’t really see any difference if I don’t set it. From the documentation (https://docs.searxng.org/admin/settings/settings_server.html), it seems like we should let the user handle this configuration.

Reply:
> How about we move them to where server.{bind_address, port} are being set?

By doing so, the user might no longer be able to edit `server.secret_key`, for which, we could introduce a new option `secretKey`?

Reply:
@drupol Sorry, your previous defaults were right. I realised this while I was trying to get the tests working on Linux and realised that making `use_default_settings = true;` mandatory by moving it along with `server.{bind_address, port}` might be problematic. Here’s why:

I was able to get the tests pass in Linux by using the following config:

```nix
{  
  services.searxng."searxng1" = {
    enable = true;
    settings = {
      # Can't use this in test as it tries to connect to search engines during init, which is not possible in the sandbox where `nix flake check` runs.
      # use_default_settings = true;

      # Keys mandatory for starting the applications
      server.secret_key = "secret";
      doi_resolvers."dummy" = "http://example.org";
      default_doi_resolver = "dummy";
    };
  };
}
```

The tests are currently failing because of `use_default_settings = true;`, which would have the search engines pre-configured. During init, `searxng` tries to connect to these engines. As the flake checks happen in a sandbox, we will have to disable this behaviour by not using `use_default_settings = true;`.

Reply:
Could you also use the ^ config in `searxng_test.nix`, the Linux test should pass after that. You can verify this by running `just test searxng` inside the devShell.